.github/workflows/pre-commit.yaml

```diff
@@ -0,0 +1,20 @@
+name: pre-commit
```

Comment:
We've had issues w/ pre-commit running in CI (I think the `artifact-definitions` repo). It's intended to be run locally before the commit happens. We can have this here, but if it causes issues (like the `aws-credentials` check) we'll need to rip it out.

Reply:
Sounds good to me! I copied the workflows the CLI has but if these cause issues we can yank em